### DISCUSSION TITLE
Mediapipe4u Package Problem

### DISCUSSION THREAD

**Discussion by yingwaima** [endink/Mediapipe4u-plugin#157] [Q&A]
Hi, everything looks great and I want to buy the license, but I can't package the file with launcher and have already installed all the SDK and Visual C++. And attached the log. Thanks a lot.

LogUObjectHash: Compacting FUObjectHashTables data took   0.50ms
LogTurnkeySupport: Selected target: MetaHuman_M4U
LogLauncherProfile: Unable to use promoted target - ../../../../../../Project/2024/unrealtest/MetaHuman_M4U/Binaries/Win64/MetaHuman_M4U.target does not exist.
LogMonitoredProcess: Running Serialized UAT: [ cmd.exe /c ""D:/Program Files/Epic Games/UE_5.2/Engine/Build/BatchFiles/RunUAT.bat"  -ScriptsForProject="D:/Project/2024/unrealtest/MetaHuman_M4U/MetaHuman_M4U.uproject" Turnkey -command=VerifySdk -platform=Win64 -UpdateIfNeeded -EditorIO -EditorIOPort=51298  -project="D:/Project/2024/unrealtest/MetaHuman_M4U/MetaHuman_M4U.uproject" BuildCookRun -nop4 -utf8output -nocompileeditor -skipbuildeditor -cook  -project="D:/Project/2024/unrealtest/MetaHuman_M4U/MetaHuman_M4U.uproject" -target=MetaHuman_M4U  -unrealexe="D:\Program Files\Epic Games\UE_5.2\E
ngine\Binaries\Win64\UnrealEditor-Cmd.exe" -platform=Win64 -installed -stage -archive -package -build -pak -iostore -compressed -prereqs -archivedirectory="D:/Project/2024/unrealtest/pack" -clientconfig=Shipping -nodebuginfo" -nocompile -nocompileuat ]
UATHelper: Packaging (Windows): Running AutomationTool...
UATHelper: Packaging (Windows): Using bundled DotNet SDK version: 6.0.302
UATHelper: Packaging (Windows): Starting AutomationTool...
UATHelper: Packaging (Windows): Parsing command line: -ScriptsForProject=D:/Project/2024/unrealtest/MetaHuman_M4U/MetaHuman_M4U.uproject Turnkey -command=VerifySdk -platform=Win64 -UpdateIfNeeded -EditorIO -EditorIOPort=51298 -project=D:/Project/2024/unrealtest/MetaHuman_M4U/MetaHuman_M4U.uproject BuildCookRun -nop4 -utf8output -nocompileeditor -skipbuildeditor -cook -project=D:/Project/2024/unrealtest/MetaHuman_M4U/MetaHuman_M4U.uproject -target=MetaHuman_M4U -unrealexe="D:\Program Files\Epic Games\UE_5.2\Engine\Binaries\Win64\UnrealEditor-Cmd.exe" -platform=Win64 -installed -stage -archive -
package -build -pak -iostore -compressed -prereqs -archivedirectory=D:/Project/2024/unrealtest/pack -clientconfig=Shipping -nodebuginfo -nocompile -nocompileuat
UATHelper: Packaging (Windows): Initializing script modules...
UATHelper: Packaging (Windows): Total script module initialization time: 0.12 s.
UATHelper: Packaging (Windows): Executing commands...
UATHelper: Packaging (Windows): Installed Sdk validity:
UATHelper: Packaging (Windows): Win64: (Status=Valid, MinAllowed_Sdk=10.0.00000.0, MaxAllowed_Sdk=10.9.99999.0, Current_Sdk=10.0.22621.0, Allowed_AutoSdk=10.0.18362.0, Current_AutoSdk=, Flags="InstalledSdk_ValidVersionExists")
UATHelper: Packaging (Windows): Scanning for envvar changes...
UATHelper: Packaging (Windows): ... done! 
UATHelper: Packaging (Windows): Cleaning Temp Paths...
UATHelper: Packaging (Windows): BUILD SUCCESSFUL
UATHelper: Packaging (Windows): Setting up ProjectParams for D:\Project\2024\unrealtest\MetaHuman_M4U\MetaHuman_M4U.uproject
UATHelper: Packaging (Windows): ********** BUILD COMMAND STARTED **********
UATHelper: Packaging (Windows): Running: D:\Program Files\Epic Games\UE_5.2\Engine\Binaries\ThirdParty\DotNet\6.0.302\windows\dotnet.exe "D:\Program Files\Epic Games\UE_5.2\Engine\Binaries\DotNET\UnrealBuildTool\UnrealBuildTool.dll" MetaHuman_M4U Win64 Shipping -Project=D:\Project\2024\unrealtest\MetaHuman_M4U\MetaHuman_M4U.uproject  D:\Project\2024\unrealtest\MetaHuman_M4U\MetaHuman_M4U.uproject -NoUBTMakefiles  -remoteini="D:\Project\2024\unrealtest\MetaHuman_M4U"  -skipdeploy  -Manifest=D:\Project\2024\unrealtest\MetaHuman_M4U\Intermediate\Build\Manifest.xml -NoHotReload -log="C:\Users\ying
w\AppData\Roaming\Unreal Engine\AutomationTool\Logs\D+Program+Files+Epic+Games+UE_5.2\UBT-MetaHuman_M4U-Win64-Shipping.txt"
UATHelper: Packaging (Windows): Log file: C:\Users\yingw\AppData\Roaming\Unreal Engine\AutomationTool\Logs\D+Program+Files+Epic+Games+UE_5.2\UBT-MetaHuman_M4U-Win64-Shipping.txt
UATHelper: Packaging (Windows): Miss GSTREAMER_1_0_ROOT_MSVC_X86_64 or GSTREAMER_ROOT environment variables.
UATHelper: Packaging (Windows): Find MediaPipe File: mediapipe4u.license
UATHelper: Packaging (Windows): Find LLM Module File: llm.conf
UATHelper: Packaging (Windows): Find Speech Module File: overlipsync\ovrlipsync_offline_model.pb
UATHelper: Packaging (Windows): Missing precompiled manifest for 'MediaPipeLLMSpeech', 'D:\Project\2024\unrealtest\MetaHuman_M4U\Plugins\MediaPipe4ULLMSpeech\Intermediate\Build\Win64\UnrealGame\Shipping\MediaPipeLLMSpeech\MediaPipeLLMSpeech.precompiled'. This module was most likely not flagged for being included in a precompiled build - set 'PrecompileForTargets = PrecompileTargetsType.Any;' in MediaPipeLLMSpeech.build.cs to override. If part of a plugin, also check if its 'Type' is correct.
UATHelper: Packaging (Windows): Took 0.7413854s to run dotnet.exe, ExitCode=6
UATHelper: Packaging (Windows): UnrealBuildTool failed. See log for more details. (C:\Users\yingw\AppData\Roaming\Unreal Engine\AutomationTool\Logs\D+Program+Files+Epic+Games+UE_5.2\UBT-MetaHuman_M4U-Win64-Shipping.txt)
UATHelper: Packaging (Windows): AutomationTool executed for 0h 0m 2s
UATHelper: Packaging (Windows): AutomationTool exiting with ExitCode=6 (6)
UATHelper: Packaging (Windows): BUILD FAILED
PackagingResults: Error: Unknown Error

[UBT-MetaHuman_M4U-Win64-Shipping.txt](https://github.com/endink/Mediapipe4u-plugin/files/14619486/UBT-MetaHuman_M4U-Win64-Shipping.txt)

**Comment by endink** [comment]
I guess you are using Unreal engie from source code (not from Launcher).

refer: https://opensource-labijie-com.translate.goog/Mediapipe4u-plugin/package/source_engine_package.html?_x_tr_sl=zh-CN&_x_tr_tl=en&_x_tr_hl=zh-CN&_x_tr_pto=wapp

**Comment by yingwaima** [comment]
Thanks a lot. The viewport can launch the cam and build successful, but the output .exe can't launch the cam. 

https://drive.google.com/file/d/1PGONFZ76vPVG_M0Hu536wBNvhmLV6dBi/view?usp=sharing

[log_0316.txt](https://github.com/endink/Mediapipe4u-plugin/files/14622973/log_0316.txt)

**Comment by endink** [comment]
Try latest version (20240524), if issue still existed, show me your blueprint here
